WordPress/openverse-frontend
# Simplify Filters to fix the SSR problem

## Fixes


Fixes #1748 by @zackkrida

## Description


This PR fixes the filters components and makes it simpler. Instead of a computed component, we select the filter component based on `v-if`. And because the conditions are mutually exclusive, there is never a case when there are two FilterGrids in the document, so there is no element `id` collision as there were previously.

I realized that if a composable has an `onMounted` hook, it runs it whenever any component that imports this composable is mounted. So, it was updating the filters visibility status in the local storage every time a component using `use-filter-visibility` mounted.  For mobile screens, we set the visibility to false in onMounted even if the localStorage was false. So, when visibility was set to true on mobile, the `onMounted` hook for the menu would run, and set the visibility back to false.

## Testing Instructions

Build and start the app using `pnpm build && pnpm start`. Go to the search page (http://localhost:8443/search/audio?q=birds), set the screen wider than mobile breakpoint and open the filters. Reload the page and check that the filters sidebar has the checkboxes inside it.
Also try making the screen wider or narrower while the filters are open. The filters should change into full screen modals on mobile and sidebar on desktop, and the desktop filters should not cover the search results.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
